src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -216,13 +247,57 @@ public String ingestDataset(
     // Fill in a default load id if the caller did not provide one in the ingest request.
     String loadTag = loadService.computeLoadTag(ingestRequestModel.getLoadTag());
     ingestRequestModel.setLoadTag(loadTag);
-    String description =
-        "Ingest from "
-            + ingestRequestModel.getPath()
-            + " to "
-            + ingestRequestModel.getTable()
-            + " in dataset id "
-            + id;
+
+    String description;
+    // Note: we are writing ingested to a bucket if specified before the flight starts so that the
+    // data does not get materialized in the flight DB. This means that we also need to edit the
+    // request to remove data that shouldn't get stored
+    if (ingestRequestModel.getFormat().equals(FormatEnum.ARRAY)) {
+      Dataset dataset = datasetDao.retrieve(UUID.fromString(id));
+      ingestRequestModel.setProfileId(
+          Optional.ofNullable(ingestRequestModel.getProfileId())
+              .orElse(dataset.getDefaultProfileId()));
+      // Create staging area if needed and get the path where the temo file will live
+      String tempFilePath =
+          jobService
+              .newJob(
+                  "Create file staging area", DatasetScratchFilePrepareFlight.class, null, userReq)
+              .addParameter(JobMapKeys.BILLING_ID.getKeyName(), ingestRequestModel.getProfileId())
+              .addParameter(JobMapKeys.DATASET_ID.getKeyName(), id)
+              .addParameter(IngestMapKeys.TABLE_NAME, ingestRequestModel.getTable())
+              .submitAndWait(String.class);
+
+      CloudPlatformWrapper cloudPlatform =
+          CloudPlatformWrapper.of(dataset.getDatasetSummary().getCloudPlatform());
+      String pathToUse;
+      if (cloudPlatform.isGcp()) {
+        pathToUse =
+            writeIngestRowsToGcpBucket(dataset, tempFilePath, ingestRequestModel.getRecords());
+      } else if (cloudPlatform.isAzure()) {
+        pathToUse =
+            writeIngestRowsToAzureStorageAccount(
+                userReq,
+                ingestRequestModel.getProfileId(),
+                dataset,
+                tempFilePath,
+                ingestRequestModel.getRecords());
+      } else {
+        throw new IllegalArgumentException("Cloud not recognized");
+      }
+      ingestRequestModel.setPath(pathToUse);
+      // Clear the json object so that it doesn't get written to the flight db
+      ingestRequestModel.getRecords().clear();
+      description = "Ingest tabular data to " + ingestRequestModel.getTable() + " in dataset " + id;
+    } else {
+      description =
+          "Ingest from "
+              + ingestRequestModel.getPath()
+              + " to "
+              + ingestRequestModel.getTable()
+              + " in dataset id "
+              + id;
```

Comment:
`String.format` would probable make this more readable.

Reply:
totally